### PR TITLE
feat: Style panel binding variable reset

### PR DESCRIPTION
**Pull request by ianxinnew**
English | [简体中文](https://github.com/opentiny/tiny-engine/blob/develop/.github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.zh-CN.md)

# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-engine/blob/develop/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Built its own designer, fully self-validated

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

### What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
  - Enhanced interactivity with added modal and reset functionality in configuration items.
  - Improved visual presentation with updated styling for modals and reset buttons.

- **Bug Fixes**
  - Ensured consistent modal positioning and visibility toggling.

- **Refactor**
  - Streamlined component imports and usage across various modules for better maintainability.
  
- **Style**
  - Adjusted styling properties such as padding, border radius, and text display for better user experience.
  - Updated color variables in the theme for improved accessibility and design consistency.

### CODE REVIEW

File: packages/common/component/ConfigItem.vue [hunk 1]
```diff
@@ -174,10 +187,13 @@ export default {
     const { t, locale } = i18n.global
 
     const verification = reactive({
+      showModal: false,
       failed: false,
       message: '',
-      hasRule: computed(() => hasRule(props.property?.required, props.property?.rules))
+      hasRule: computed(() => hasRule(props.property?.required, props.property?.rules)),
+      defaultValue: props.property?.widget?.props?.modelValue
```
Comment (chilingling) [inline packages/common/component/ConfigItem.vue]: 这里为什么需要 `reactive` ？

File: packages/common/component/ConfigItem.vue [hunk 2]
```diff
@@ -468,10 +484,23 @@ export default {
...
+    const { setPosition } = useModalMask()
+
+    const openReset = (event) => {
+      if (showBindState.value) {
+        setPosition(event)
+        verification.showModal = true
+      }
+    }
+    const reset = () => {
+      variable.value.remove()
```
Comment (chilingling) [inline packages/common/component/ConfigItem.vue:495]: 假如配置了 `"bindState": false`，然后，通过修改 schema 的方式给该属性绑定变量，这里是不是就会有 bug ？

File: packages/common/component/ConfigItem.vue [hunk 3]
```diff
@@ -624,6 +657,12 @@ export default {
         border-bottom: 1px dashed;
       }
     }
+    .pro-bind {
+      padding: 2px;
+      color: var(--ti-lowcode-meta-config-item-bind-color);
+      background: var(--ti-lowcode-meta-config-labe-item-bind-bg), 0.08;
```
Comment (chilingling) [inline packages/common/component/ConfigItem.vue]: 这个是什么语法？

File: packages/common/component/ModalMask.vue [hunk 4]
```diff
@@ -17,14 +17,28 @@ const modal = reactive({
...
   const setPosition = (event) => {
+    const target = event.target
+    // 获取触发事件元素的高度
+    const elementHeight = target.offsetHeight
+    // 获取触发事件元素距离顶部的高度
+    const elementTop = target.getBoundingClientRect().top + window.scrollY
+    // 获取页面总高度
+    const pageHeight = document.documentElement.scrollHeight
+    const elementBottom = pageHeight - elementTop - elementHeight
     const panelWidth = window.getComputedStyle(document.body).getPropertyValue('--base-right-panel-width')
```
Comment (chilingling) [inline packages/common/component/ModalMask.vue:30]: 这里的 `--base-right-panel-width` 是仅适用于右侧面板的，但是现在放置到 common 了。
可以考虑：
1. 修改相关配置，使得该方法更通用
2. 限制仅 setting 面板可使用

File: packages/settings/styles/src/js/useStyle.js [hunk 5]
```diff
@@ -248,12 +248,15 @@ watch(
     if (!classNameList) {
       return
     }
-
-    const matchStyles = Object.values(state.styleObject).filter(
-      (value) => value.pureSelector === classNameList && value.mouseState === mouseState
-    )
-    const style = matchStyles.length ? matchStyles[0].rules : {}
-    state.style = style
+    if (state.styleObject[classNameList].pureSelector) {
```
Comment (chilingling) [inline packages/settings/styles/src/js/useStyle.js]: 这样改把原来的逻辑都改掉了...

**Review by chilingling — CHANGES_REQUESTED** [review]

File: packages/theme/light/base.less [hunk 6]
```diff
@@ -270,4 +271,7 @@
   --ti-lowcode-base-success-color: #5cb300;
   --ti-lowcode-base-prompt-color: #1476ff;
   --ti-lowcode-base-invalidate-color: #ebebeb;
+  // 遮罩朦板
+  --lowcode-base-mode-mask-background: #fff;
```
Comment (hexqi) [inline packages/theme/light/base.less]: 命名不规范，然后base变量应该放在base包中，并且这里更应该是common？

File: packages/theme/light/metaComponent.less [hunk 7]
```diff
@@ -24,8 +24,9 @@
 .properties-item {
   --ti-lowcode-meta-config-item-error-tips-color: var(--ti-lowcode-base-error-color);
   --ti-lowcode-meta-config-item-label-color: var(--ti-lowcode-base-text-color);
-  --ti-lowcode-meta-config-item-bind-color: var(--ti-lowcode-base-gray-0);
-  --ti-lowcode-meta-config-item-bind-bg: var(--ti-lowcode-base-primary-color-2);
+  --ti-lowcode-meta-config-item-bind-color: var(--ti-lowcode-base-text-color-2);
+  --ti-lowcode-meta-config-item-bind-bg: var(--ti-lowcode-base-text-color-5);
+  --ti-lowcode-meta-config-labe-item-bind-bg: rgba(20, 118, 255, 0.08);
```
Comment (hexqi) [inline packages/theme/light/metaComponent.less:29]: 要使用色值变量

**Pull request closed without merging**